### PR TITLE
log doppelganger attestation signature; rm withState.HashedBeaconState uses

### PR DESCRIPTION
**Pull request by tersec**
- log attestation signature during doppelganger detection to enable further tracing of it
- remove 2 of 4 `withState`-based users of `HashedBeaconState`, which just wanted the state root
- remove some unused templates/variables/types
- remove build-time reference to empty test https://github.com/status-im/nimbus-eth2/blob/unstable/tests/test_sync_protocol.nim which is already tested for compilation by being included from nimbus_beacon_node and sync/request_manager. It was creating a build-time warning due to not using its imported module.

### CODE REVIEW

File: tests/all_tests.nim [hunk 1]
```diff
@@ -29,7 +29,6 @@ import # Unit test
   ./test_state_transition,
   ./test_statediff,
   ./test_sync_manager,
-  ./test_sync_protocol,
```
Comment (arnetheduck) [inline tests/all_tests.nim:32]: ? 

this should still be there

Comment (tersec) [reply tests/all_tests.nim:32]: Why? It tests nothing.

Comment (arnetheduck) [reply tests/all_tests.nim:32]: ie either remove the file completely or leave the all_tests thing in there

Comment (arnetheduck) [reply tests/all_tests.nim:32]: I think we actually have a few legacy smoke tests like this from before we had a binary to compile - they can safely be removed, but we shouldn't have garbage files in the tests repo that are _not_ included in `make test` in general

Comment (tersec) [reply tests/all_tests.nim:32]: Removed: https://github.com/status-im/nimbus-eth2/pull/2608/commits/a018ce7c062462f04bc1c3e96f07dffbcf1317ca

**Pull request merged**